### PR TITLE
[2] GraphiQL x Vite v5

### PR DESCRIPTION
**Pull request by dimaMachina**
closes https://github.com/graphql/graphiql/pull/2925 https://github.com/graphql/graphiql/pull/2371/ https://github.com/graphql/graphiql/pull/2638/ https://github.com/graphql/graphiql/issues/2184

I propose to cut v4 alpha release and include this PR and new tabs looks PR https://github.com/graphql/graphiql/pull/3644

changed exports

`graphiql/graphiql.js` removed non-minified umd bundle
`graphiql/graphiql.min.js` => `graphiql/dist/index.umd.js`
`graphiql/graphiql.css` => `graphiql/style.css` (added `exports` field in `package.json`)

todos:

- [ ] add changeset, major, describe changed exports
- [ ] search in code base to replace `import 'graphiql/style.css';`
- [x] remove webpack deps https://github.com/graphql/graphiql/pull/3681

### CODE REVIEW

File: packages/graphiql/resources/renderExample.js [hunk 1]
```diff
@@ -83,6 +83,7 @@ function getSchemaUrl() {
 // how you can customize GraphiQL by providing different values or
 // additional child elements.
 const root = ReactDOM.createRoot(document.getElementById('graphiql'));
+const graphqlVersion = GraphiQL.GraphQL.version;
```
Comment (dimaMachina) [inline packages/graphiql/resources/renderExample.js:86]: found how to easily get version of `graphql-js` without importing `package.json` or injecting code

File: packages/graphiql/vite.config.mts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import commonjs from 'vite-plugin-commonjs';
+
+const umdConfig = defineConfig({
+  define: {
+    'process.env.NODE_ENV': `"${process.env.NODE_ENV}"`,
+  },
+  // To bundle `const { createClient } = require('graphql-ws')` in `createWebsocketsFetcherFromUrl` function
+  plugins: [commonjs()],
+  build: {
+    minify: true,
```
Comment (dimaMachina) [inline packages/graphiql/vite.config.mts:13]:
```suggestion
```
can be removed to use default value

File: packages/graphiql/vite.config.mts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+});
+
+const esmConfig = defineConfig({
+  build: {
+    minify: false,
+    sourcemap: true,
+    lib: {
+      entry: 'src/index.ts',
+      fileName: 'index',
+      formats: ['cjs', 'es'],
```
Comment (dimaMachina) [inline packages/graphiql/vite.config.mts:42]:
```suggestion
      formats: ['cjs', 'es'],
```
do we really need a bundle of `cjs` version for the next major? or just left esm?

Comment (klippx) [reply packages/graphiql/vite.config.mts:42]: It's quite a big step to get rid of cjs, if it doesn't cost you much to keep outputting cjs?

Comment (dimaMachina) [reply packages/graphiql/vite.config.mts:42]: with dropping `cjs` we can bundle `graphql-toolkit` to esm only too and fix https://github.com/graphql/graphiql/issues/3668

**Pull request merged**